src/QMCHamiltonians/RandomRotationState.h

```diff
@@ -0,0 +1,62 @@
+//////////////////////////////////////////////////////////////////////////////////////
+// This file is distributed under the University of Illinois/NCSA Open Source License.
+// See LICENSE file in top directory for details.
+//
+// Copyright (c) 2022 QMCPACK developers.
+//
+// File developed by: Mark Dewing, mdewing@anl.gov, Argonne National Laboratory
+//
+// File created by: Mark Dewing, mdewing@anl.gov, Argonne National Laboratory
+//////////////////////////////////////////////////////////////////////////////////////
+
+
+#ifndef QMCPLUSPLUS_RANDOM_ROTATION_STATE_H
+#define QMCPLUSPLUS_RANDOM_ROTATION_STATE_H
+
+#include "Configuration.h"
+#include "Utilities/RandomGenerator.h"
+#include <array>
+
+namespace qmcplusplus
+{
+
+// The angular values for a random rotation matrix
+// phi : 0 to 2*pi
+// psi : 0 to 2*pi
+// cth : -1 to 1  (cth = cos(theta))
+class RandomRotationState
+{
+public:
+  using RealType   = QMCTraits::RealType;
+  using TensorType = QMCTraits::TensorType;
+
+  RandomRotationState() : rng_vals_({0.0, 0.0, 0.0}) {}
+  RandomRotationState(RandomGenerator& rng) : rng_vals_({rng(), rng(), rng()}) {}
+
+  TensorType getRandomRotationMatrix() const
```

Comment:
I think this should just be a function that takes an RNG and returns a random rotation matrix.